spec/lib/common/schema_validator_spec.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'common/schema_validator'
+
+describe Common::SchemaValidator do
+  let(:response_body) do
+    { 'data' =>
+    [{ 'id' => '121133',
+       'identifier' => [{ 'system' => 'http://med.va.gov/fhir/urn/vaos/vista/983/appointment/id',
+                          'value' => '999;20220827.094500' }],
+       'kind' => 'clinic',
+       'status' => 'cancelled',
+       'service_type' => 'optometry',
+       'patient_icn' => '1012846043V576341',
+       'location_id' => '983',
+       'clinic' => '455',
+       'start' => '2022-08-27T10:30:00Z',
+       'cancelation_reason' =>
+       { 'coding' =>
+         [{ 'system' => 'http://terminology.hl7.org/CodeSystem/appointment-cancellation-reason',
+            'code' => 'pat',
+            'display' => 'The appointment was cancelled by the patient' }] },
+       'cancellable' => false,
+       'extension' => { 'cc_location' => { 'address' => {} },
+                        'vista_status' => ['CANCELLED BY PATIENT', 'FUTURE'] } }] }
+  end
+
+  context 'when response matches schema' do
+    it 'does not log an error' do
+      expect(Rails.logger).not_to receive(:error)
+      Common::SchemaValidator.new(response_body.to_json, 'modules/vaos/app/schemas/appointments.json').validate
+    end
+  end
+
+  context 'when response has additional attributes' do
```

Comment:
These error cases are mostly testing to confirm that the validations work as expected and they do appear to.